controller/user.js

```diff
@@ -171,29 +171,31 @@ async function verifyAccount(req, res) {
   });
 }
 
-async function update(req, res, next) {
-  const id = getIdParam(req);
-  if (!req.body.id) {
-    throw new errors.FcError(errors.MISSING_USER_ID);
-  } else {
-    models.User.update(req.body, {
-      where: { id: id },
-    })
-      .then((rowsUpdate) => {
-        if (rowsUpdate[0] == 0) {
-          throw new errors.FcError(errors.USER_NOT_FOUND);
-        } else {
-          models.User.findByPk(id).then((user) => {
-            res.status(statusCode.SUCCESS).json({
-              code: 0,
-              msg: "",
-              data: { user: buildUserJson(user) },
-            });
-          });
-        }
-      })
-      .catch(next);
+async function update(req, res) {
+  const user_id = !req.params.id ? req.user.id : getIdParam(req);
+  const user = await models.User.findByPk(user_id);
+  if (!user) {
+    throw new errors.FcError(errors.USER_NOT_FOUND);
   }
+  
+  const rowsChange = await models.User.update({
+    full_name: req.body.full_name,
+    school_name: req.body.school_name
+  }, {
+    where: { id: user_id },
+  });
+
+  if (!rowsChange[0]) {
```

Comment:
Not really, nếu không Update được nó thuộc về system error vì em đã check được là user này exist rồi